waku/v2/node/rest/server.nim

```diff
@@ -0,0 +1,86 @@
+{.push raises: [Defect].}
+
+import
+  std/[os, times],
+  std/typetraits,
+  stew/[byteutils, io2],
+  stew/shims/net,
+  chronicles, chronos,
+  metrics, metrics/chronos_httpserver,
+  bearssl,
+  presto
+
+
+proc getRouter(allowedOrigin: Option[string]): RestRouter =
+  # TODO: Review this `validate` method. Check in nim-presto what is this used for.
+  proc validate(key: string, value: string): int =
+    ## This is rough validation procedure which should be simple and fast,
+    ## because it will be used for query routing.
+    1
+
+  RestRouter.init(validate, allowedOrigin = allowedOrigin)
+
+
+type RestServerConf* = object 
+      cacheSize*: Natural ## \
+        ## The maximum number of recently accessed states that are kept in \
+        ## memory. Speeds up requests obtaining information for consecutive  
+        ## slots or epochs.
+
+      cacheTtl*: Natural ## \
+        ## The number of seconds to keep recently accessed states in memory
+
+      requestTimeout*: Natural ## \
+        ## The number of seconds to wait until complete REST request will be received
+
+      maxRequestBodySize*: Natural ## \
+        ## Maximum size of REST request body (kilobytes)
+
+      maxRequestHeadersSize*: Natural ## \
+        ## Maximum size of REST request headers (kilobytes)
+
+proc default(T: type RestServerConf): RestServerConf =
+  RestServerConf(
+      cacheSize: 3,
+      cacheTtl: 60,
+      requestTimeout: 0,
+      maxRequestBodySize: 16_384,
+      maxRequestHeadersSize: 64
+  )
+
+
+template init*(T: type RestServerRef,
```

Comment:
Just wondering: What is the reason for using a template here instead of a proc?

Reply:
I just copied and adapted this server init method from the nimbus codebase.

Reply:
oh! There might be a reason for doing this which I don't see :sweat_smile: